pronto_mia/lib/ui/views/deployment_plan/edit/deployment_plan_edit_view.dart

```diff
@@ -27,90 +38,100 @@ class DeploymentPlanEditView extends StatelessWidget
+              firstDate: deploymentPlan != null
+                  ? deploymentPlan.availableFrom
+                  : DateTime.now(),
+              lastDate: DateTime.now().add(const Duration(days: 365)),
+              dateMask: 'dd.MM.yyyy HH:mm',
+              dateLabelText: 'Gültig ab*',
+            ),
+            DateTimePicker(
+              type: DateTimePickerType.dateTime,
+              controller: availableUntilController,
+              firstDate: deploymentPlan != null
+                  ? deploymentPlan.availableUntil
+                  : DateTime.now(),
+              lastDate: DateTime.now().add(const Duration(days: 365)),
+              dateMask: 'dd.MM.yyyy HH:mm',
+              dateLabelText: 'Gültig bis*',
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              crossAxisAlignment: CrossAxisAlignment.baseline,
+              textBaseline: TextBaseline.ideographic,
+              children: [
+                TextButton(
+                  onPressed: () => handlePdfUpload(model),
+                  child: const Text('Datei hochladen'),
+                ),
+                const SizedBox(width: 8.0),
+                Expanded(
+                  child: TextField(
+                    controller: pdfPathController,
+                    readOnly: true,
+                    onTap: model.openPdf,
+                    style: const TextStyle(
+                      color: Colors.blue,
+                      decoration: TextDecoration.underline,
                     ),
                   ),
-                  if (model.pdfUpload != null)
-                    TextButton(
-                      onPressed: model.openPdf,
-                      child: const Text('Anzeigen'),
-                    )
-                ],
-              ),
-              const SizedBox(height: 16.0),
-              SizedBox(
-                  width: double.infinity,
-                  height: 40.0,
-                  child: ElevatedButton(
-                    onPressed: model.submitForm,
-                    child: const Text('Erstellen'),
-                  )),
-              if (model.validationMessage != null) ...[
-                const SizedBox(height: 8.0),
-                Text(
-                  model.validationMessage,
-                  style: const TextStyle(color: Colors.red),
                 ),
               ],
-            ])),
+            ),
+          ],
+          primaryButton: ButtonSpecification(
+            title: 'Speichern',
+            onTap: model.submitForm,
+            isBusy: model.busy(model.editBusyKey),
+          ),
+          secondaryButton: (() {
+            if (deploymentPlan != null) {
+              return ButtonSpecification(
+                title: 'Löschen',
+                onTap: model.removeDeploymentPlan,
+                isBusy: model.busy(model.removeBusyKey),
+                backgroundColor: Colors.red,

```

Comment:
I would adjust the ButtonSpecification to have a boolean argument `isDestructive` which will then set the color accordingly instead of defining colors all over the place.

Reply:
Added isDestructive in favor of backgroundColor.